lib/matplotlib/pyplot.py

```diff
@@ -480,88 +479,65 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
     in the matplotlibrc file.
     """
 
-    if figsize is None:
-        figsize = rcParams['figure.figsize']
-    if dpi is None:
-        dpi = rcParams['figure.dpi']
-    if facecolor is None:
-        facecolor = rcParams['figure.facecolor']
-    if edgecolor is None:
-        edgecolor = rcParams['figure.edgecolor']
-
     allnums = get_fignums()
     next_num = max(allnums) + 1 if allnums else 1
-    figLabel = ''
+    fig_label = ''
     if num is None:
         num = next_num
     elif isinstance(num, str):
-        figLabel = num
-        allLabels = get_figlabels()
-        if figLabel not in allLabels:
-            if figLabel == 'all':
+        fig_label = num
+        all_labels = get_figlabels()
+        if fig_label not in all_labels:
+            if fig_label == 'all':
                 cbook._warn_external(
-                    "close('all') closes all existing figures")
+                    "close('all') closes all existing figures.")
             num = next_num
         else:
-            inum = allLabels.index(figLabel)
+            inum = all_labels.index(fig_label)
             num = allnums[inum]
     else:
         num = int(num)  # crude validation of num argument
 
-    figManager = _pylab_helpers.Gcf.get_fig_manager(num)
```

Comment:
Slight preference for `fig_manager` or `figmanager`. IMO just `manager` is a bit too generic. Applies throughout the PR.

Reply:
There's quite a few other functions that already use `manager` (pause(), sca(), subplot_tool()), and I'd say the context makes things quite clear.

Reply:
I was particularly stumbling over `manager = _pylab_helpers.Gcf.get_active()` which is really opaque. But that's maybe more due to bad naming of `get_active()`.

Disregarding context and what may be used elsewhere I still find `figmanager` better than just `manager`. IMO the three extra letters rather add to the understandablility of the code. I won't hold the PR if others approve, but I do not approve myself.

Reply:
ok.